Source: nodejs-mobile/nodejs-mobile@38bcb111d024fd1feaa059947d074e6848415c7e
Source: nodejs-mobile/nodejs-mobile@88c3d79d7915aad1020f693241a1ba403b6b2ff8

diff --git a/deps/uv/src/unix/darwin.c b/deps/uv/src/unix/darwin.c
index c26a1faccbcc4b9c4f87f8635792a46e1e738fe6..bea4676d89e031dfd503fe09a4f59e6889663520 100644
--- a/deps/uv/src/unix/darwin.c
+++ b/deps/uv/src/unix/darwin.c
@@ -20,6 +20,7 @@
 
 #include "uv.h"
 #include "internal.h"
+#include <TargetConditionals.h>
 
 #include <assert.h>
 #include <stdint.h>
@@ -206,12 +207,14 @@ int uv_cpu_info(uv_cpu_info_t** cpu_infos, int* count) {
   }
 
   cpuspeed = 0;
+#if !TARGET_OS_IPHONE
   size = sizeof(cpuspeed);
   sysctlbyname("hw.cpufrequency", &cpuspeed, &size, NULL, 0);
   if (cpuspeed == 0)
     /* If sysctl hw.cputype == CPU_TYPE_ARM64, the correct value is unavailable
      * from Apple, but we can hard-code it here to a plausible value. */
     cpuspeed = 2400000000U;
+#endif
 
   if (host_processor_info(mach_host_self(), PROCESSOR_CPU_LOAD_INFO, &numcpus,
                           (processor_info_array_t*)&info,
diff --git a/deps/uv/uv.gyp b/deps/uv/uv.gyp
index 760a3cdb0019d1c1c030d77fd7e9233714dba78a..c258d5e31a34fea26ba6fc5ba83f25d320f52791 100644
--- a/deps/uv/uv.gyp
+++ b/deps/uv/uv.gyp
@@ -291,8 +291,22 @@
           'defines': [
             '_GNU_SOURCE',
           ],
-          'sources': [
-            '<@(uv_sources_android)',
+          'target_conditions': [
+            ['_toolset=="target"', {
+              'sources': [
+                '<@(uv_sources_android)',
+              ],
+            }],
+            ['_toolset=="host" and host_os=="mac"', {
+              'sources': [
+                '<@(uv_sources_apple)',
+              ],
+            }],
+            ['_toolset=="host" and host_os=="linux"', {
+              'sources': [
+                '<@(uv_sources_linux)',
+              ],
+            }],
           ],
           'link_settings': {
             'libraries': [ '-ldl' ],
